helm-system-packages-portage.el

```diff
@@ -119,51 +105,37 @@ The caller can pass the list of EXPLICIT packages to avoid re-computing it."
                    (buffer-string)))
    explicit))
 
-(defun helm-system-packages-portage-cache-names ()
-  "Cache all package names."
-  (with-temp-buffer
-    (process-file "eix" nil t nil "--only-names")
-    (buffer-string)))
-
-(defcustom helm-system-packages-portage-column-width 36
-  "Column at which descriptions are aligned, excluding a double-space gap."
+(defun helm-system-packages-portage-cache (display-list)
+  "Cache all package names with descriptions."
+  (let* ((raw (with-temp-buffer
+                (process-file "env" nil '(t nil) nil "EIX_LIMIT=0" "OVERLAYS_LIST=none" "PRINT_COUNT_ALWAYS=never" "eix" "--format" "<category>/<name>\t<description>\n")
+                (sort-lines nil (point-min) (point-max))
+                (buffer-string)))
+         (paired (mapcar (lambda (x) (let ((l (split-string x "\t"))) (cons (car l) (cadr l)))) (split-string raw "\n")))
+         (names (mapcar 'car paired))
+         (descriptions (mapcar (lambda (x) (concat (car x) (make-string (max (- helm-system-packages-column-width (length (car x))) 0) ? ) (cdr x))) paired)))
+    (helm-system-packages--cache-set names descriptions display-list "portage")))
+
+(defcustom helm-system-packages-portage-column-width 40
+  "Column at which descriptions are aligned, excluding a double-space gap.
+If nil, then use `helm-system-packages-column-width'."
   :group 'helm-system-packages
   :type 'integer)
 
-(defun helm-system-packages-portage-cache-descriptions ()
-  "Cache all package names with descriptions."
-  (with-temp-buffer
-    ;; `eix' can format the output while `apt-cache search' cannot.  Thus we
-    ;; tell `eix' to use the same formatting as `apt-cache' so that we can
-    ;; re-use its code.
-    ;; TODO: Can eix pad in the format string just like `expac' does?
-    ;; TODO: Or output straight to Elisp?
```

Comment:
What do you think of outputting straight to Elisp?  Could be dangerous, but could also be simpler and faster (if speed is a problem).
If you think this is still relevant, please keep the "TODO" note.

Reply:
Although the `eix` documentation that exists is fairly sparse, I couldn't find any reference to padding in the format string anywhere. It would certainly help significantly if this was possible. As far as outputting directly to Elisp, I assume you mean writing a series of S-expressions and then using `read`? I agree that this is fairly dangerous, and we'd have to perform a post-processing step anyway, since `eix` doesn't have any ability to escape strings AFAIK.

Reply:
OK.  Feel free to comment on the TODO in the code.